hvac/tests/integration_tests/api/auth/test_ldap.py

```diff
@@ -0,0 +1,356 @@
+import logging
+import os
+from unittest import TestCase
+
+from ldap_test import LdapServer
+from parameterized import parameterized
+
+from hvac import exceptions
+from hvac.tests import utils
+
+LDAP_URL = 'ldap://ldap.hvac.network'
+LDAP_GROUP_NAME = 'vault-users'
+LDAP_USER_NAME = 'somedude'
+LDAP_USER_PASSWORD = 'hvacrox'
+LDAP_BASE_DC = 'hvac'
+LDAP_BASE_DN = 'dc={dc},dc=network'.format(dc=LDAP_BASE_DC)
+LDAP_BIND_DN = 'cn=admin,{base_dn}'.format(base_dn=LDAP_BASE_DN)
+LDAP_BIND_PASSWORD = 'notaverygoodpassword'
+LDAP_USERS_DN = 'dc=users,{base_dn}'.format(base_dn=LDAP_BASE_DN)
+LDAP_GROUPS_OU = 'groups'
+LDAP_GROUPS_DN = 'ou={ou},{base_dn}'.format(ou=LDAP_GROUPS_OU, base_dn=LDAP_BASE_DN)
+LDAP_LOGIN_USER_DN = 'uid={username},{users_dn}'.format(username=LDAP_USER_NAME, users_dn=LDAP_USERS_DN)
+LDAP_ENTRIES = [
+    {
+        'objectclass': 'domain',
+        'dn': LDAP_USERS_DN,
+        'attributes': {
+            'dc': 'users'
+        }
+    },
+    {
+        'objectclass': ['inetOrgPerson', 'posixGroup', 'top'],
+        'dn': LDAP_LOGIN_USER_DN,
+        'attributes': {
+            'uid': LDAP_USER_NAME,
+            'userPassword': LDAP_USER_PASSWORD
+        }
+    },
+    {
+        'objectclass': 'organizationalUnit',
+        'dn': LDAP_GROUPS_DN,
+        'attributes': {
+            'ou': 'groups',
+        }
+    },
+    {
+        'objectclass': 'groupOfNames',
+        'dn': 'cn={cn},{groups_dn}'.format(cn=LDAP_GROUP_NAME, groups_dn=LDAP_GROUPS_DN),
+        'attributes': {
+            'cn': LDAP_GROUP_NAME,
+            'member': LDAP_LOGIN_USER_DN,
+        }
+    },
+]
+
+
+def load_test_cert():
+    """Load test cert for use when testing Ldap.configure's `certificate` parameter.
+
+    :return: Test certificate contents
+    :rtype: str | unicode
+    """
+    test_data_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', '..', '..', '..', 'test')
```

Comment:
These '..'s smell funny - a helper method somewhere (hvac/tests/utils.py? or a new hvac/tests/test_files.py?) would be nice.

```python
from hvac.tests import test_files
test_cert = test_files.get_file_contents('server-cert.pem')
```

`get_file_contents(name)` could call a `get_filename(name)` that joins with a `get_base_dir()`.
removes some boilerplate, and later if test files are moved around, would only need to update `get_base_dir()`.

Reply:
@dbwpe: Does 79fc92c1244743d0c7adf0d56800efac931c5242 cover what you had in mind?

Reply:
Then I read your actual comment and bc57308 is closer to the mark...